File: omc/android/app/src/main/java/com/pureenee/model/ProdottoOrdine.java
package com.pureenee.model;

import io.objectbox.annotation.Backlink;
import io.objectbox.annotation.Entity;
import io.objectbox.annotation.Id;
import io.objectbox.relation.ToOne;

@Entity
public class ProdottoOrdine {

    @Id
    private long id;


    ToOne<ProdottoDeal> prodottoDeal;

    ToOne<Ordine> ordine;

    private double quantitativoEffettivo;

    private double quantitativoPrevisto;


    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public double getQuantitativoEffettivo() {
        return quantitativoEffettivo;
    }

    public void setQuantitativoEffettivo(double quantitativoEffettivo) {
        this.quantitativoEffettivo = quantitativoEffettivo;
    }

    public double getQuantitativoPrevisto() {
        return quantitativoPrevisto;
    }

    public void setQuantitativoPrevisto(double quantitativoPrevisto) {
        this.quantitativoPrevisto = quantitativoPrevisto;
    }

    public double getCresta() {
        return this.quantitativoPrevisto - this.quantitativoEffettivo;
    }

    public void setProdottoDeal(ProdottoDeal prodottoDeal){
        this.prodottoDeal.setTarget(prodottoDeal);
    }

    public Ordine getOrdine(){
        return this.ordine.getTarget();
    }

    public void setOrdine(Ordine ordine){
        this.ordine.setTarget(ordine);
    }

    public ProdottoDeal getProdottoDeal(){
        return this.prodottoDeal.getTarget();
    }

}

File: omc/lib/model/prodotto_ordine.dart

import 'package:omc/model/prodotto_deal.dart';

class ProdottoOrdine {

  final int id;
  double quantitativoEffettivo;
  double quantitativoPrevisto;
  ProdottoDeal prodottoDeal;

  ProdottoOrdine(
      this.id,
      this.quantitativoEffettivo,
      this.quantitativoPrevisto,
      this.prodottoDeal
  );

  factory ProdottoOrdine.fromJson(Map<String, dynamic> json) {
    return ProdottoOrdine(
      json['id'],
      json['quantitativoEffettivo'],
      json['quantitativoPrevisto'],
      ProdottoDeal.fromJson(json['prodottoDeal'])
    );
  }

  Map<String, dynamic> toJson() => {
    'id': id,
    'quantitativoEffettivo': quantitativoEffettivo,
    'quantitativoPrevisto': quantitativoPrevisto,
    'prodottoDeal': prodottoDeal.toJson(),
  };

  //setQuantitativoPrevisto(double quantitativoPrevisto){
  //  this.quantitativoPrevisto = quantitativoPrevisto;
  //}


}